lbry/lbry/dht/node.py

```diff
@@ -245,3 +227,71 @@ def accumulate_peers(self, search_queue: asyncio.Queue,
                          asyncio.Queue, asyncio.Task]:
         q = peer_queue or asyncio.Queue(loop=self.loop)
         return q, self.loop.create_task(self._accumulate_search_junction(search_queue, q))
+
+    async def _maintain_connection(self, seed_peers: typing.List['KademliaPeer'], timeout: float = None):
+        elapsed = 0
+        while True:
+            if timeout and elapsed > timeout:
```

Comment:
This timeout should be removed - it will cause join_network to give up.

Reply:
also, if join_network were to have a timeout the top level call could instead be wrapped in `asyncio.wait_for`

Reply:
The timeout is put here so that both timed (for fixed seeds) and infinite (for restored peers) loops are callable from `join_network`.

Reply:
And the timeout is not for the whole call, just the state of waiting for peers to respond...

Reply:
The `join_network` loop keeps running and will cause the node to re-join if it loses connectivity and the routing table is emptied. The refresh loop could have still been running, and thus the persisted peer can have been cleared out as well by the time connectivity is restored.

The `peers` list can be populated at the start of the loop with the persisted peers, or if there aren't any with the fixed peers. Then if by the end of the loop the join wasn't yet successful the fixed peers can be added to the `peers` list if they weren't already added. This would cause the join network loop to try persisted peers and possibly the fixed peers until the next cycle of the refresh loop, and afterward fall back to only the fixed peers if the join still hasn't been successful.